extension-client-exec.md

```diff
@@ -0,0 +1,33 @@
+# `exec` extension to LSP
+
+The `exec` extension allows the server to send a command (e.g. `git blame`) to the client for execution, and the client responds with the stdout of the command.
+
+### `exec` Request
+
+_Request_:
+* method: 'xexec'
+* params: `ExecParams` defined as follows:
+```typescript
+interface ExecParams {
+    /**
+     * The name of the command to run
+     */
+    name: string;
+    /**
+     * The arguments to the command
+     */
+    arguments: string;
```

Comment:
This should in general match the API for `exec` in programming language core libs. `arguments` needs to be `string[]` (so it supports a single arg and doesn't just always pass it to a shell for evaluation).